libslide/src/bignum/add.rs

```diff
@@ -0,0 +1,135 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::Bignum;
+use std::mem;
+use std::ops;
+
+impl ops::Add for Bignum {
+    type Output = Bignum;
+
+    fn add(self, rhs: Bignum) -> Bignum {
+        let is_neg: bool = rhs.is_neg && self.is_neg;
+
+        // this assumes the signs are both positive for now
+        let mut carry: u8 = 0;
+        let mut lhs_size: usize = self.dec.len();
+        let mut rhs_size: usize = rhs.dec.len();
+        let mut res_dec: Vec<u8> = Vec::new();
+
+        // 1. Handle Decimal
+
+        // swap if neccessary to make lhs smaller. Initialize big vec as the bigger vector
+        let big_vec = if lhs_size > rhs_size {
+            mem::swap(&mut lhs_size, &mut rhs_size);
+            self.dec.clone()
+        } else {
+            rhs.dec.clone()
+        };
+
+        // compute the decimal addition up to the size of the small vec
+        let mut res: u8;
+        for i in (0..lhs_size).rev() {
+            res = self.dec[i] + rhs.dec[i] + carry;
+            res_dec.push(res % 10);
```

Comment:
I don't think we should be pushing on an empty vec. There is a really easy way to solve this and the rest of your code on L36-44, which is to just initialize res_dec as the larger of the two vectors (self.dec or rhs.dec).

Reply:
What is wrong with pushing an empty vec?

Reply:
Way more allocations